app/common/data/interfaces/collections.py

```diff
@@ -402,6 +403,21 @@ def create_group(
     parent: Optional[Group] = None,
     presentation_options: QuestionPresentationOptions | None = None,
 ) -> Group:
+    # If this group is nested, ensure it meets rules for nesting groups
+    # This is a safety check as we don't allow users to create nested groups when these rules aren't met
+    if parent:
+        if not parent.is_allowed_nested_groups:
```

Comment:
optional nit: The naming of this just caused me a 🤏  bit of confusion - it sounds like that attribute is answering whether or not you can create a nested group at all (which we have two logical rules for: # of levels of nesting, and the parent being same page). This attribute only checks the level of nesting.